src/main/java/org/opensearch/commons/destination/message/LegacyCustomWebhookMessage.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.opensearch.commons.destination.message;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+
+import org.apache.http.client.methods.HttpPatch;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.opensearch.common.Strings;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+
+/**
+ * This class holds the content of an CustomWebhook message
+ */
+public class LegacyCustomWebhookMessage extends LegacyBaseMessage {
+
+    private final String message;
+    private final String url;
+    private final String scheme;
+    private final String host;
+    private final String method;
+    private final int port;
+    private String path;
+    private Map<String, String> queryParams;
+    private Map<String, String> headerParams;
+    private final String userName;
+    private final String password;
+
+    private LegacyCustomWebhookMessage(
+        final LegacyDestinationType destinationType,
+        final String destinationName,
+        final String url,
+        final String scheme,
+        final String host,
+        final Integer port,
+        final String path,
+        final String method,
+        final Map<String, String> queryParams,
+        final Map<String, String> headerParams,
+        final String userName,
```

Comment:
we don't support username/password (it was not supported by earlier code as well). please remove.

Reply:
Removed since ISM is not using it and it might cause confusion on what is supported.